### PR TITLE
Day 1 build system hotfixes

### PR DESCRIPTION
**Pull request by gtaharaedmonds**
### Summary

Various build system fixes from meeting 1.

### CODE REVIEW

File: environment/scripts/install_stm32cubemx.py
Comment (gtaharaedmonds) [inline environment/scripts/install_stm32cubemx.py:1]: Changes in this file are all from linting

File: firmware/CMakeLists.txt [hunk 1]
```diff
@@ -94,7 +107,7 @@ function(cube_mx_code_generation
     add_custom_command(
         OUTPUT ${IOC_FILE}.md5
         ${LOG4J_PROPERTIES}
-        COMMAND python3
+        COMMAND ${XVFB_PREFIX} python3
```
Comment (Lucien950) [inline firmware/CMakeLists.txt:110]: I assume that the build system error `xvfb-run: error: Xvfb failed to start` is coming from here.

**Review by Lucien950 — CHANGES_REQUESTED** [review]

File: firmware/CMakeLists.txt (hunk 1 above)
Comment (gtaharaedmonds) [reply firmware/CMakeLists.txt]: Yeah I'm trying to fix this now

Comment (gtaharaedmonds) [reply firmware/CMakeLists.txt]: it doesn't happen when building locally

**Review by will-chaba — APPROVED** [review]
Seems good to me. If we're going native linux/wsl don't we need the Pipfile back?

**Comment by gtaharaedmonds** [comment]
> Seems good to me. If we're going native linux/wsl don't we need the Pipfile back?

we could but for development personally I don't think python virtual environments add a lot of value
the situation where you need a different version of package for project A vs. B is pretty rare 
also pipenv is very slow

**Pull request merged**